partner_tlr_update/models/res_partner.py

```diff
@@ -119,35 +133,43 @@ def update_from_xml(self, xml, match_name):
                             if field[1] == 'country_id.name':
                                 country = value_id.id
                 elif field[1] in self:
+                    _logger.info("field[1] in self")
                     data[field[1]] = elem.text
 
         if data:
             self.write(data)
 
     @api.model
     def parse_xml_tjansteleverantor_data(self, xml):
-        root = etree.XML(xml)
+        root = etree.XML(xml.encode())
         self.update_from_xml(root, 'organization')
+        _logger.info("UPDATE FROM XML")
         contact_persons = root.find(
-            ".//ns107:kontaktpersonLista", namespaces=root.nsmap)
+            ".//ns107:kontaktpersonLista", namespaces={'ns107': 'http://arbetsformedlingen.se/datatyp/tjansteleverantor/tjansteleverantor/v15'})
+        _logger.info("CONTACT PERSONS FOUND")
         if contact_persons is not None:
+            _logger.info("CONTACT PERSONS NOT NONE")
             self.update_contact_person(contact_persons)
 
     @api.model
     def parse_xml_organisationsnummer_data(self, xml):
+        #_logger.info("XML: %s" % xml)
         root = etree.fromstring(self.clearing_xml(xml))
+        _logger.info("ROOT: %s" % root)
         self.update_subsidiary(root)
 
     @api.multi
     def update_contact_person(self, xml):
-        person_id = xml.find(".//ns25:kontaktpersonId", namespaces=xml.nsmap)
+        person_id = xml.find(".//ns25:kontaktpersonId", namespaces={'ns25':'http://arbetsformedlingen.se/datatyp/tjansteleverantor/kontaktperson/v17'})
```

Comment:
should this be hardcoded?

Reply:
Preferably not, but xml.nsmap only returns the namespace of the root tag so this is the best I could come up with.